culinary-canvas/capstone-project/frontend/my_capstone_project/src/Notification.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import { useAuth } from "./RedirectToAuthentication";
+import {useNavigate} from 'react-router-dom';
+import { Link } from "react-router-dom";
+const backendAddress = import.meta.env.VITE_BACKEND_ADDRESS;
+import { getSocket } from "./socket";
+
+function Notification(){
+    const [notifications, setNotifications] = useState([]);
+    const [loading, setLoading] = useState(true)
+    const {isAuthenticated, user} = useAuth();
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        const socket = getSocket();
+        if(isAuthenticated && user){
+            if(!user){
+                navigate('/')
+            }
+            const fetchNotifications = async() => {
+                try{
+                    const response = await fetch(`${backendAddress}/notifications/${user.id}`, {
+                        method: 'GET',
+                        credentials: 'include',
+                    });
+                    const data = await response.json();
+                    setNotifications(data)
+                }catch(error){
+                    console.error('Error fetching notifications: ', error)
+                }finally{
+                    setLoading(false);
+                }
+            };
+
+            fetchNotifications();
+            socket.on('connect', () => {
+                socket.emit('suscribe', user.id)
+            });
+            socket.on('notifications', (data) => {
+                setNotifications(data)
+            });
+            socket.on('notifications', (notification) => {
+                setNotifications((prev) => [notification, ...prev])
+            });
+            socket.on('disconnect', () => {
+                console.log('Disconnected from web socket server')
+            })
+            socket.on('connect_error', (error) => {
+                console.error('Connection error: ', error)
+            })
+            socket.on('reconnect_attempt', () => {
+                console.log('Attempting to reconnect...')
+            })
+            return () => {
+                socket.off('connect');
+                socket.off('notifications');
+                socket.off('disconnect');
+                socket.off('connect_error');
+                socket.off('reconnect_attempt');
+            };
+        }
+
+    }, [user, isAuthenticated]);
+
+
+    const markAsRead = async(notificationId) => {
+        try{
+            await fetch(`${backendAddress}/notifications/read`, {
+                method: 'POST',
+                headers: {'Content-Type' : 'application/json'},
+                body: JSON.stringify({notificationId}),
+                credentials: 'include'
+            });
+            setNotifications(notifications.map(n => n.id === notificationId ? {...n, read: true} : n))
```

Comment:
Small improvement suggestion: Instead of using map, you can use findIndex to see if the specific notification exists and modify just that one element's 'read' to true.